sql/moz-fx-data-shared-prod/activity_stream/dataset_metadata.yaml

```diff
@@ -10,3 +10,4 @@ workgroup_access:
   members:
   - workgroup:mozilla-confidential
   - workgroup:pocket/prefect
+  - workgroup:pocket/external
```

Comment:
The docs mention: 

> If a table does not have workgroup_access metadata specified, its access will be determined by the workgroup_access on its dataset. Datasets are guaranteed to have workgroup_access metadata.

So would it make sense to have 

```yaml
labels:
  authorized: true
workgroup_access:
  - role: roles/bigquery.dataViewer
    members:
      - workgroup:pocket/external
```
in the `dataset_metadata.yaml` and removing the manually added metadata files? Or does this cause problems here? Also does the same apply to `authorized: true` being applied to the tables?

Reply:
The docs are a bit confusing here. Dataset `workgroup_access` and configuration are not inherited by underlying tables i.e. dataset-level `workgroup_access` is not a default value for a table's `workgroup_access`. Table ACLs are only deployed when they are specified, and are additive to dataset access, so the docs are saying that if table ACLs are not specified a table will be accessible only by its dataset ACLs, and that these ACLs will always exist.

`labels: authorized: true` is only for authorized views currently. Authorized datasets exist but we don't currently support them in general: https://mozilla-hub.atlassian.net/browse/DSRE-947, but that metadata would look something like https://github.com/mozilla/bigquery-etl/pull/3177/files.

The issue with granting this access at the dataset level is that the original purpose of `activity_stream_bi` as a separate dataset was to avoid granting blanket access to all tables in the `activity_stream` dataset: https://bugzilla.mozilla.org/show_bug.cgi?id=1616895#c1. Any dataset-level grant will apply to all tables regardless of table configuration. See the top-level review comment for more on that.

In practice it would be much simpler to switch to [uniform dataset access](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27924789/Data+Access+Workgroups#DataAccessWorkgroups-Exampledatasetwithuniformspecificworkgroupaccess(creatingadatasetthatisaccessibletoaspecificworkgroup)) which is what this line does. This appears to be how we implemented pocket/prefect access (DSRE-532). I don't know what prefect is but this may be the time to revisit the policy decision on what data can/should be accessible to pocket's various systems. However, we don't really have an owner of this data that I'm aware of to make these kinds of policy decisions, and I probably should have sorted that out before implementing DSRE-532.

All to say: if we can switch to a single dataset grant for `pocket/external` then we can greatly simplify our configuration, at the expense of mode having access to a broader set of data than it had before. This would require somebody to make an ownership/policy decision and I don't know who that person is.

Reply:
Additional note: this requires a corresponding PR to https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/prod/envs/prod/bigquery-new/namespaces.auto.tfvars.json#L11 so that metadata is synchronized between ops logic and bqetl.

Reply:
> This would require somebody to make an ownership/policy decision and I don't know who that person is.
I brought this up to Rob yesterday, who will look for a new owner

Reply:
> All to say: if we can switch to a single dataset grant for pocket/external then we can greatly simplify our configuration, at the expense of mode having access to a broader set of data than it had before. This would require somebody to make an ownership/policy decision and I don't know who that person is.

Sounds like the original concern from [Bug 1616895](https://bugzilla.mozilla.org/show_bug.cgi?id=1616895#c1) was around granting Mode access to data containing `client_id` values. Would that become the case if we switch to a single dataset grant?
If **not**, we can safely switch to that approach. 
If **so**, I think we should land the more complex config as it's currently being proposed, and open a separate conversation about whether sharing `client_id`s with Mode should be allowed. As part of that conversation, we can also track where that requirement stems from (looks like an RRA from quite a few years ago).

Are other policy concerns that need to be addressed here?

Reply:
The `events` and `sessions` tables under `activity_stream` do look to have telemetry `client_id`. So I guess we'll go ahead without switching to uniform data access for now.

Reply:
@mreid-moz @scholtzan How can I help further the conversation on whether Mode is allowed to use `client_id`? Kenny Long and I would love to see if we can revisit that RRA from a long time ago.

Reply:
Also, when you refer to the "complex config", I understand that means granting Mode access at the table/view level vs. the schema level, but does that have any bearing on whether Mode gets access to the client_id?

Reply:
> The events and sessions tables under activity_stream do look to have telemetry client_id. 

> Sounds like the original concern from [Bug 1616895](https://bugzilla.mozilla.org/show_bug.cgi?id=1616895#c1) was around granting Mode access to data containing client_id values.

This feels like a policy blocker to me. I can review the PR as if we're requiring granular access, but the granular access includes `events` which contains `client_id`. I agree with @kirill-demtchouk that furthering the conversation on whether Mode is allowed to use `client_id` is the way to proceed here.

So the options as I see them are:

1. have a policy conversation that ends in allowing `client_id` access for Mode, in which case this PR can be simplified to use only uniform dataset access
2. have a policy conversation that ends in retaining the same policy about `client_id` access for Mode, in which case this PR should not include tables that have that column. If derived versions e.g. `select * EXCEPT (client_id)` of those tables are desirable, they could be added to `activity_stream_bi` as authorized views

Reply:
@kirill-demtchouk Yes, I think we should revisit that decision. Do you have access to the original RRA doc (linked in [Bug 1486889](https://bugzilla.mozilla.org/show_bug.cgi?id=1486889))? If so please share it with me. If not, please [request a new RRA](https://infosec.mozilla.org/guidelines/risk/rapid_risk_assessment) and include me in the meeting. Thanks!

Reply:
@mreid-moz Done! I shared 1486889 with you. Is there anything else you need from me to revisit the RRA?

Reply:
@kirill-demtchouk I still haven't been able to access the original RRA document linked from Bug 1486889. Please request a new RRA, and add me to the attendee list.